### PR TITLE
Verify that SR-IOV resource injector can inject downward API volume

### PR DESCRIPTION
**Pull request by fedepaol**
Note: the deletion of networks was not implemented here as it's being implemented in https://github.com/openshift/sriov-tests/pull/18

### CODE REVIEW

File: conformance/operator_test.go [hunk 1]
```diff
@@ -265,6 +267,113 @@ var _ = Describe("operator", func() {
 				Expect(err).To(HaveOccurred())
 			})
 		})
+		Context("Resource Injector", func() {
+			It("Should inject downward api volume", func() {
```
Comment (kononovn) [inline conformance/operator_test.go]: Could you add test id between these lines as a comment ?

**Review by kononovn — COMMENTED** [review]
Excellent job!

File: conformance/operator_test.go [hunk 2]
```diff
@@ -265,6 +267,114 @@ var _ = Describe("operator", func() {
...
+				Eventually(func() corev1.PodPhase {
+					runningPod, err = clients.Pods(namespaces.Test).Get(created.Name, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					return runningPod.Status.Phase
+				}, 3*time.Minute, time.Second).Should(Equal(corev1.PodRunning))
+
+				var downwardVolume *corev1.Volume
+				for _, v := range runningPod.Spec.Volumes {
+					if v.Name == "podnetinfo" {
+						downwardVolume = v.DeepCopy()
```
Comment (SchSeba) [inline conformance/operator_test.go:618]: nit: lets break here

File: conformance/operator_test.go [hunk 3]
```diff
@@ -265,6 +267,114 @@ var _ = Describe("operator", func() {
...
+				Expect(downwardVolume.DownwardAPI).ToNot(BeNil(), "Downward api not found in volume")
+				Expect(downwardVolume.DownwardAPI.Items).To(SatisfyAll(
+					ContainElement(corev1.DownwardAPIVolumeFile{
+						Path: "labels",
+						FieldRef: &corev1.ObjectFieldSelector{
+							APIVersion: "v1",
+							FieldPath:  "metadata.labels",
+						},
+					}), ContainElement(corev1.DownwardAPIVolumeFile{
+						Path: "labels",
```
Comment (SchSeba) [inline conformance/operator_test.go]: I think this one should be `annotations`

File: conformance/operator_test.go [hunk 4]
```diff
@@ -265,6 +267,114 @@ var _ = Describe("operator", func() {
...
+						FieldRef: &corev1.ObjectFieldSelector{
+							APIVersion: "v1",
+							FieldPath:  "metadata.labels",
```
Comment (SchSeba) [inline conformance/operator_test.go]: also this one

File: conformance/operator_test.go [hunk 5]
```diff
@@ -265,6 +267,114 @@ var _ = Describe("operator", func() {
...
+					},
+					Spec: sriovv1.SriovNetworkSpec{
+						ResourceName:     "apivolresource",
+						IPAM:             `{"type":"host-local","subnet":"10.10.10.0/24","rangeStart":"10.10.10.171","rangeEnd":"10.10.10.181","routes":[{"dst":"0.0.0.0/0"}],"gateway":"10.10.10.1"}`,
+						NetworkNamespace: namespaces.Test,
+					}}
+				err = clients.Create(context.Background(), sriovNetwork)
+				Expect(err).ToNot(HaveOccurred())
+
+				pod := &corev1.Pod{
```
Comment (SchSeba) [inline conformance/operator_test.go]: I think later you can use the pod package from this PR https://github.com/openshift/sriov-tests/pull/18

**Review by SchSeba — CHANGES_REQUESTED** [review]
Almost ready

File: conformance/operator_test.go (hunk 1 above)
Comment (fedepaol) [reply conformance/operator_test.go]: Done

File: conformance/operator_test.go (hunk 3 above)
Comment (fedepaol) [reply conformance/operator_test.go]: Thanks for catching it!

File: conformance/operator_test.go (hunk 5 above)
Comment (fedepaol) [reply conformance/operator_test.go]: Yes, after everything is settled we can do a round of refactoring

**Review by SchSeba — APPROVED** [review]
/lgtm

**Review by SchSeba — APPROVED** [review]
/lgtm

**Pull request merged**